llvm-external-projects/iree-dialects/lib/Dialect/LinalgTransform/IR/StructuredTransformOpsExt.cpp

```diff
@@ -376,6 +376,25 @@ static scf::ForOp findSingleForOpDefiningAll(ValueRange range) {
   return forOp;
 }
 
+/// Find the op that defines all values in the range.
+static Operation *findSingleOpDefiningAll(ValueRange range) {
+  Operation *op = nullptr;
+  for (Value value : range) {
+    if (auto currentSourceOp = value.getDefiningOp()) {
+      if (!op || op == currentSourceOp) {
+        op = currentSourceOp;
+        continue;
+      }
+      LLVM_DEBUG(DBGS() << "different source op when replacing one op\n");
```

Comment:
I suppose this needs to `return nullptr` here. Otherwise it will return the defining op of the first value even in the case of different sources.

Reply:
Good point, here and above.